src/errors.rs

```diff
@@ -32,29 +32,56 @@ impl From<anyhow::Error> for EncodeError {
 }
 
 #[derive(Debug, Error)]
-#[error("Decode error occurred: {inner}")]
-pub struct DecodeError {
-    inner: anyhow::Error,
+pub enum DecodeError {
+    #[error("Invalid MAC address")]
+    InvalidMACAddress,
```

Comment:
This is perhaps a bit more tedious with a questionable benefit, but you can store the received payload size. Then the error message could become `Invalid MAC address: expected 6 bytes, received {payload} bytes`.

The same goes for `InvalidIPAddress` and other similar errors.